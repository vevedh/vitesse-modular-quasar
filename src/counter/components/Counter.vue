<script setup lang="ts">
const props = defineProps<{
  initial: number
}>()
const { count, inc, dec } = useCounter(props.initial)
</script>

<template>
  <div class="flex items-center space-x-5 justify-center mt-10 mb-20">
    <button class="btn" data-test="dec" @click="dec()">
      -
    </button>
    <span>{{ count }}</span>
    <button class="btn" data-test="inc" @click="inc()">
      +
    </button>
  </div>
</template>
