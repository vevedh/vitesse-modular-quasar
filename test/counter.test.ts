import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import Counter from '~/counter/components/Counter.vue'

describe('Counter.vue', () => {
  it('should render', () => {
    const wrapper = mount(Counter, { props: { initial: 10 } })
    expect(wrapper.text()).toContain('10')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should increment by 1', async () => {
    const wrapper = mount(Counter, { props: { initial: 0 } })
    expect(wrapper.text()).toContain('0')

    expect(wrapper.find('[data-test="inc"]').exists()).toBe(true)

    await wrapper.get('[data-test="inc"]').trigger('click')

    expect(wrapper.text()).toContain('1')
  })

  it('should decrement by 1', async () => {
    const wrapper = mount(Counter, { props: { initial: 1 } })
    expect(wrapper.text()).toContain('1')

    expect(wrapper.find('[data-test="dec"]').exists()).toBe(true)

    await wrapper.get('[data-test="dec"]').trigger('click')

    expect(wrapper.text()).toContain('0')
  })
})
