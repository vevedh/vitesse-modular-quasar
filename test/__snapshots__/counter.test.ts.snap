// Vitest Snapshot v1

exports[`Counter.vue > should render 1`] = `"<div class=\\"flex items-center space-x-5 justify-center mt-10 mb-20\\"><button class=\\"btn\\" data-test=\\"dec\\"> - </button><span>10</span><button class=\\"btn\\" data-test=\\"inc\\"> + </button></div>"`;
